app/models/category_list.rb

```diff
@@ -78,7 +78,12 @@ def find_categories
     if SiteSetting.fixed_category_positions
       @categories = @categories.order(:position, :id)
     else
-      @categories = @categories.includes(:latest_post).order("posts.created_at DESC NULLS LAST").order('categories.id ASC')
+      allowed_category_ids = @categories.pluck(:id) << nil
```

Comment:
This line is a bit odd to me. It's taking the array of `id`s and then adding a value for `nil`? Why is that necessary?

Reply:
`nil` is necessary to include categories without any topics, equivalent for `.where("topics.category_id IN (?) OR topics.category_id IS null", allowed_category_ids)`

Reply:
Looks like a trick that is worth adding a comment about